src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this[this.length - 1] === undefined) {
+      this[this.length - 1] = '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
```

Comment:
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
        result += this[i];
      }
```

Reply:
But if element is null or undefined it should be replaced with ' ' . 

Expected: ",,1,,0,NaN,"
    Received: "1,0,NaN,,,"

      75 | test(`Should work when array has null as a first element`, () => {
      76 |   expect([null, null, 1, undefined, 0, NaN, ''].join2(','))
    > 77 |     .toBe(',,1,,0,NaN,');
         |      ^